### PR TITLE
fix(script): decode create2 constructor args

### PR DESCRIPTION
**Pull request by colinnielsen**
## Motivation

While working on metropolis, I found that create2 constructor arguments were not being deserialized and put in the `tx.arguments` array of the broadcast artifacts

## Solution

Simply include a 32 byte offset (to remove the salt) and re-use the logic from CREATE transaction decoding

### CODE REVIEW

File: crates/forge/bin/cmd/script/transaction.rs [hunk 1]
```diff
@@ -133,60 +130,44 @@ impl TransactionWithMetadata {
...
+
+                        let params =
+                            constructor.inputs.iter().map(|p| p.kind.clone()).collect::<Vec<_>>();
+
+                        // the constructor args start after bytecode
+                        let constructor_args = &creation_code[info.code.len()..];
+
+                        if let Ok(arguments) = abi::decode(&params, constructor_args) {
+                            self.arguments = Some(arguments.iter().map(format_token_raw).collect());
+                            println!("{:?}", self.arguments);
```
Comment (mattsse) [inline crates/forge/bin/cmd/script/transaction.rs:166]: this is new, should we keep this?

File: crates/forge/bin/cmd/script/transaction.rs [hunk 2]
```diff
@@ -133,60 +130,44 @@ impl TransactionWithMetadata {
...
-                                let bytecode = hex::encode(&data.0);
-                                (signature, bytecode)
-                            };
-
-                            let params = constructor
+            if let Some(info) = contracts.get(&address) {
+                // constructor args are postfixed to creation code
+                // and create2 transactions are prefixed by 32 byte salt
+                let contains_constructor_args = if is_create2 {
+                    data.len() - 32 > info.code.len()
```
Comment (mattsse) [inline crates/forge/bin/cmd/script/transaction.rs:137]: -.-
nice find

**Review by mattsse — APPROVED** [review]
cool, nice find!

one question re new println!

File: crates/forge/bin/cmd/script/transaction.rs (hunk 1 above)
Comment (colinnielsen) [reply crates/forge/bin/cmd/script/transaction.rs]: Oops - my apologies, I'll remove this

File: crates/forge/bin/cmd/script/transaction.rs (hunk 2 above)
Comment (colinnielsen) [reply crates/forge/bin/cmd/script/transaction.rs:137]: ty :)

**Review by mattsse — APPROVED** [review]
lgtm

**Pull request merged**